### ISSUE TITLE
NullReferenceException when node is null

### ISSUE THREAD

**Issue by chertby** [dotnetrdf/dotnetrdf#516]
Can I add null check for arguments in [TreeIndexedTripleCollection.cs](https://github.com/dotnetrdf/dotnetrdf/blob/main/Libraries/dotNetRdf/Core/TreeIndexedTripleCollection.cs)?

`NullReferenceException` is too general and doesn't make it easier to find bugs.

![image](https://user-images.githubusercontent.com/15835359/187770963-0430916d-96b1-4a32-9ef1-1949eff54284.png)

**Comment by kal** [comment]
You are right, all of the public methods that expect a non-null argument should validate the arguments throw an `ArgumentNullException` if they are null.
